freetype.cmake

```diff
@@ -0,0 +1,18 @@
+set(FT_DISABLE_BROTLI ON CACHE STRING "Disable Brotli" FORCE)
+set(FT_DISABLE_BZIP2 ON CACHE STRING "Disable bzip2" FORCE)
+set(FT_DISABLE_HARFBUZZ ON CACHE STRING "Disable HarfBuzz" FORCE)
```

Comment:
The subdirectory stuff showing up in the option list is always annoying. Let's put mark_as_advanced for all those

Reply:
Done. I have also marked as advanced another one we don't modify.

While I was at it, I modified those lines as `option()`, and we don't force them anymore, we just set the default value we want, and this takes precedence over FreeTypes's own defaults, but still allow us to tweak them if we want.

The only forced FreeType option is `FT_DISABLE_ZLIB` (option to use builtin FreeType zlib) because it is always ON with NaCl and we already rely on `PREFER_EXTERNAL_LIBS` instead when building the engine.